src/main/java/mate/academy/spring/boot/controller/CategoryController.java

```diff
@@ -0,0 +1,56 @@
+package mate.academy.spring.boot.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import mate.academy.spring.boot.dto.book.BookDtoWithoutCategoryIds;
+import mate.academy.spring.boot.dto.category.CategoryDto;
+import mate.academy.spring.boot.mapper.BookMapper;
+import mate.academy.spring.boot.service.BookService;
+import mate.academy.spring.boot.service.CategoryService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
+@RequestMapping("/categories")
+@RequiredArgsConstructor
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookMapper bookMapper;
+    private final BookService bookService;
+
+    @PostMapping
+    @PreAuthorize("hasRole('ADMIN')")
+    public CategoryDto createCategory(@RequestBody @Valid CategoryDto categoryDto) {
```

Comment:
```suggestion
    public CategoryDto createCategory(@RequestBody @Valid CategoryRequestDto categoryDto) {
```

create CategoryRequestDto withou id field